### ISSUE TITLE
Feature: create Insight pages from Github star lists

### ISSUE THREAD

**Issue by isabensusan** [open-sauced/app#2314]
### Type of feature

🍕 Feature

### Current behavior

When creating a new Insight page, we should let users paste a link to a Github star list.

### What are Github star lists? ([Github Docs](https://docs.github.com/en/get-started/exploring-projects-on-github/saving-repositories-with-stars))
It's like a bookmark folder for projects you've starred. So for example, when you star a repo, there's a dropdown menu that allows you to save that repo to a list:
<img width="509" alt="image" src="https://github.com/open-sauced/app/assets/27322879/080c4167-5b59-41d2-b1fc-aad383c77270">

Users can show those lists on their user profiles or keep them private. They get displayed as a recollection of repositories you've saved to that list:
![Rewatch GIF Recording - 2023-12-12 at 2 53 17 PM](https://github.com/open-sauced/app/assets/27322879/9d448a66-cd06-4216-8f36-1bbbb394a9a5)

### Use Case
If a user has been saving repositories to star lists, it's because they are interested in them for a reason. By being able to syncronize a list of saved repos to create an insight page, users will be able to access data on contributions and activity for those repositories.

### Suggested solution

TBD - Needs Design

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by isabensusan** [comment]
tldr: blocked due to necessary API endpoint not existing.

While researching the existence of a Star List API endpoint on the Github API we found [this discussion](https://github.com/orgs/community/discussions/8293) where the Github team states that the endpoint is not available. Because the discussion is 2 years old by now and no updates have been given, we think this is not currently on Github's roadmap.

<img width="950" alt="image" src="https://github.com/open-sauced/app/assets/27322879/6daf1253-f1f0-4fe0-8603-d3367b27fc38">
